feat(#26): Storages speed rate limit

randreev1321 opened the pull request:
Changes:
* Added new option `limits` to main and job configs files to set limits.
* Added `limits.disk` option to set the write speed limit to a disc

Refs: #26

randreev1321 committed:
feat(#26): Disk rate limit

Changes:
* Added new option `limits` to main and job configs files to set limits.
* Added `limits.disk` option to set the write speed limit to a disc

Refs: #26

randreev1321 committed:
feat(#26): Storages rate limit

Changes:
 - Storage and job initializations have been modified in order to include a rate limit for storages.
 - Added new options to set storages rate limit
 - Renamed option to set disk rate limit in config

 Refs: #26

borisershov commented on modules/backup/desc_files/desc_files.go:
```diff
@@ -228,7 +231,7 @@ func (j *job) DoBackup(logCh chan logger.LogRecord, tmpDir string) error {
 
 		startTime := time.Now()
-		if err = targz.Tar(tgt.path, tmpBackupFile, false, tgt.gzip, tgt.saveAbsPath, tgt.excludes); err != nil {
+		if err = targz.Tar(tgt.path, tmpBackupFile, false, tgt.gzip, tgt.saveAbsPath, j.diskRateLimit, tgt.excludes); err != nil {
```
Better to use structure when transferred more than 3 arguments

borisershov commented on modules/storage/sftp/sftp.go:
```diff
@@ -41,23 +43,23 @@ type Params struct {
 }
 
-func Init(name string, params Params) (*SFTP, error) {
+func Init(name string, opts Opts, rl int64) (*SFTP, error) {
```
Why the `name` and `rl` arguments are outside the `opts` structure?

randreev1321 replied:
Because type conversion is used during initialisation. This point can be put in the refactoring task.

randreev1321 committed:
feat(#26): Review changes

Changes:
 - Fixed rate limit definition function
 - Used pointers for rate limit settings
 - Used Opts struct for Tar function

Refs: #26
